File: 0049-group-anagrams/0049-group-anagrams.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        if (strs.Length == 0) return new List<IList<string>>();
        
        Dictionary<string, List<string>> ans = new Dictionary<string, List<string>>();
        foreach (string s in strs) {
            char[] ca = s.ToCharArray();
            Array.Sort(ca);
            string key = new string(ca);
            
            if (!ans.ContainsKey(key)) {
                ans[key] = new List<string>();
            }
            
            ans[key].Add(s);
        }
        
        return ans.Values.ToList<IList<string>>();
    }
}

File: 0049-group-anagrams/0049-group-anagrams.rs
use std::collections::HashMap;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut map: HashMap<String, Vec<String>> = HashMap::new();
        for s in strs {
            let mut chars: Vec<char> = s.chars().collect();
            chars.sort();
            let key = chars.into_iter().collect::<String>();
            map.entry(key).or_default().push(s);
        }
        map.into_iter().map(|(_, v)| v).collect()
    }
}
